ui/src/utils/react-utils.tsx

```diff
@@ -68,6 +68,43 @@ export function useLoaderWithStatus<TEntity>(
     return [entity, loaderStatus];
 }
 
+
+export function useLoaderWithTimer<TEntity>(
+    setEntity: (entity: TEntity | undefined) => void,
+    loadFunc: () => Promise<TEntity> | undefined,
+    deps: unknown[],
+    status?: boolean,
+    anyFailed?: boolean,
+    setWaitingAfterFail?: (status: boolean) => void,
```

Comment:
Tämä on vähän hassu parametri. Täältä ei kannata kutsua enää kutsujaa takaisinpäin (erikoinen flow) vaan mielummin palauttaa promise, joka kertoo mitä tapahtui, vastaavasti kuin tuossa ylemmässä funktiossa. 

Itse asiassa, tämän timerin voisi varmaan jopa laittaa optional parametriksi tuohon ylempään, koska koodi tekee hyvin pitkälti saman muutoin.

Reply:
Viittaatko ylempään funktioon useLoader? Sehän ei palauta Promisea vaan ottaa sellaisen sisään, kuten tämäkin. Tässä siis annetaan parametreina myös kaksi higher order funktioita, (joista toinen on optionaalinen) ylemmältä komponentilta, joiden tila muuttuu. 

UseLoaderilla ei ole sisäistä timeriä ja ajattelin, että nämä kaksi olisi ollut hyvä erottaa selkeyden vuoksi. Ehkä pitäisi keskustella vielä yhdessä @solita-tiandu-zhao @juuseko @ksurakka-solita ja @arirahikkala  kanssa miten tällainen kannattaisi hoitaa.

Reply:
intervallin arvo on hyvä muuttaa parametriksi ja minulta unohtui tehdä. Hyvä huomio.